DigitalLearningSolutions.Web/Attributes/AllowedExtensionsAttribute.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace DigitalLearningSolutions.Web.Attributes
+{
+    using System.ComponentModel.DataAnnotations;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.AspNetCore.Http;
+
+    public class AllowedExtensionsAttribute : ValidationAttribute
+    {
+        private readonly string[] extensions;
+        private readonly string? errorMessage;
+
+        public AllowedExtensionsAttribute(string[] allowedExtensions, string? errorMessage = null)
+        {
+            this.extensions = allowedExtensions;
+            this.errorMessage = errorMessage;
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            if (value is IFormFile file)
```

Comment:
I feel like if it isn't a file it should fail, but also that shouldn't ever come up.

Reply:
I can't think of a way that would come up other than a developer putting this attribute on something that isn't a property for a file upload. I'll add some documentation to it to make it obvious.

Reply:
I guess we could always only return Success if the allowed extensions contains the file extensions, but I don't mind too much given that we control where this is used. Will leave it up to you :)

Reply:
I think leaving it as it is is the best option. Currently if a developer puts it on the wrong property type, it just won't do anything, always returning Success because it isn't an IFormFile. If we change it to fail on non-IFormFile values and it gets put on the wrong type of property, it'll cause unexpected errors because the validation will always fail (which if it somehow made it through testing/release would cause issues).
I think the always success if incorrect property type is safer.